app/cdap/components/ColumnInsights/Components/ColumnDetails/index.tsx

```diff
@@ -0,0 +1,153 @@
+    Array.isArray(DATATYPE_OPTIONS) &&
+    DATATYPE_OPTIONS?.length &&
+    DATATYPE_OPTIONS.filter((each) => each.value === columnType?.toLowerCase());
+  const [dataTypeValue, setDataTypeValue] = useState<string>();
+  const classes = useStyles();
+  const [canEdit, setCanEdit] = useState<Boolean>(false);
+  const [inputValue, setInputValue] = useState<string>(columnName);
+  const [invalidInput, setInvalidInput] = useState<Boolean>(false);
+
+  const checkForInvalidInput = (renamedString: string) => {
+    if (!/^\w+$/.test(renamedString)) {
+      setInvalidInput(true);
+    } else {
+      setInvalidInput(false);
+    }
+  };
+
+  const handleDataTypeChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setDataTypeValue(e.target.value);
+    dataTypeHandler(e.target.value);
+  };
+
+  const editHandler = () => {
+    setCanEdit(true);
+  };
+
+  const onChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    checkForInvalidInput(e.target.value);
+    setInputValue(e.target.value);
+  };
+
+  const onBlurEvent = (e: React.FocusEvent<HTMLInputElement>) => {
+    if (invalidInput) {
+      setCanEdit(true);
+    } else {
+      setCanEdit(false);
+      if (e.target.value !== columnName && !invalidInput) {
+        renameColumnNameHandler(columnName, e.target.value);
+      }
+    }
+  };
+
+  const onEnter = (e) => {
+    if (e.target.value !== columnName && !invalidInput && e.keyCode === 13) {
+      renameColumnNameHandler(columnName, e.target.value);
+      setCanEdit(false);
+    }
+  };
+  return (
+    <section className={classes.columnInsightsTopSection}>
+      <div className={classes.columnNameEdit}>
+        {canEdit ? (
+          <input
+            value={inputValue}
+            onBlur={(e) => onBlurEvent(e)}
+            onChange={(e) => onChangeHandler(e)}
+            onKeyDown={onEnter}
+          />
+        ) : (
+          <Typography className={classes.columnInsightsColumnName}>{inputValue}</Typography>
+        )}
+        <Box>
+          <EditIcon onClick={editHandler} className={classes.editIcon} />
+        </Box>
+      </div>
+      {invalidInput && (
+        <div>
+          <Typography className={classes.invalidInput} component="span" variant="body1">
+            {T.translate(`${PREFIX}.invalidInputErrorMessage`).toString()}
+          </Typography>
+        </div>
+      )}
+
+      <InputSelect
+        classes={{
+          icon: classes.selectIconStyles,
+          select: classes.selectStyles,
+        }}
+        className={classes.selectFieldStyles}
+        optionClassName={{ root: classes.optionStyles }}
+        defaultValue={defaultValueProvided[0]?.value}
+        value={dataTypeValue}
+        onChange={(e) => handleDataTypeChange(e)}
+        options={DATATYPE_OPTIONS}
+        fullWidth={false}
+      />
+      <section className={classes.columnInsightsDetailsWrapper}>
+        <div className={classes.columnInsightsDetailsCountSection}>
+          <Typography
```

Comment:
Please check if we can create a styled component and reuse it

Reply:
Added common component.
const SimpleLabel = styled(Typography)`
  &&& {
    font-size: ${(props) => props.fontSize ?? 14}px;
    line-height: 150%;
    font-weight: 400;
    color: ${(props) => props.color ?? `${grey[900]}`};
  }
`;